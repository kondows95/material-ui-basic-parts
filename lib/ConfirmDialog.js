import React from 'react';
import { Box, Dialog, DialogActions, DialogTitle } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';
import { getButtonLoadingStyle, getConfirmDialogStyle } from './styles';
import MyButton from './MyButton';
var useStyles = makeStyles(function (theme) { return ({
    progress: getButtonLoadingStyle(),
    dialog: getConfirmDialogStyle(theme),
}); });
var ConfirmDialog = function (props) {
    var classes = useStyles();
    var formatMessage = useIntl().formatMessage;
    return (React.createElement(Box, { className: classes.dialog },
        React.createElement(Dialog, { open: props.open, onClose: props.onClose, "aria-labelledby": "alert-dialog-title", "aria-describedby": "alert-dialog-description" },
            React.createElement(DialogTitle, { id: "alert-dialog-title", "data-testid": "alert-dialog-title" }, formatMessage({ id: 'MUIBP.confirmDialogMessage', defaultMessage: 'Are you sure?' })),
            React.createElement(DialogActions, null,
                React.createElement(MyButton, { label: formatMessage({ id: 'MUIBP.cancel', defaultMessage: 'Cancel' }), onClick: props.onClose, "data-testid": "confirm-cancel" }),
                React.createElement(MyButton, { label: formatMessage({ id: 'MUIBP.ok', defaultMessage: 'OK' }), onClick: props.onConfirmed, "data-testid": "confirm-ok", color: "primary", autoFocus: true })))));
};
export default ConfirmDialog;
