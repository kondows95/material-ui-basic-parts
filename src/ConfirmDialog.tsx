import React from 'react';
import { Box, Dialog, DialogActions, DialogTitle, Theme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';
import { getButtonLoadingStyle, getConfirmDialogStyle } from './styles';
import MyButton from './MyButton';

const useStyles = makeStyles((theme: Theme) => ({
    progress: getButtonLoadingStyle(),
    dialog: getConfirmDialogStyle(theme),
}));

interface Props {
    open: boolean;
    onConfirmed: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    onClose: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const ConfirmDialog: React.FC<Props> = (props) => {
    const classes = useStyles();
    const { formatMessage } = useIntl();

    return (
        <Box className={classes.dialog}>
            <Dialog
                open={props.open}
                onClose={props.onClose}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
            >
                <DialogTitle id="alert-dialog-title" data-testid="alert-dialog-title">
                    {formatMessage({ id: 'MUIBP.confirmDialogMessage', defaultMessage: 'Are you sure?' })}
                </DialogTitle>
                <DialogActions>
                    <MyButton
                        label={formatMessage({ id: 'MUIBP.cancel', defaultMessage: 'Cancel' })}
                        onClick={props.onClose}
                        data-testid="confirm-cancel"
                        color="primary"
                    />
                    <MyButton
                        label={formatMessage({ id: 'MUIBP.ok', defaultMessage: 'OK' })}
                        onClick={props.onConfirmed}
                        data-testid="confirm-ok"
                        color="primary"
                        autoFocus
                    />
                </DialogActions>
            </Dialog>
        </Box>
    );
};
export default ConfirmDialog;
