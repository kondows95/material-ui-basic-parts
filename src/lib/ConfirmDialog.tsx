import React from 'react';
import { Dialog, DialogActions, DialogTitle } from '@material-ui/core';
import { useIntl } from 'react-intl';
import MyButton from './MyButton';

const ignoreKeys = ['open', 'onConfirmed', 'onClose'];
interface Props {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
    open: boolean;
    onConfirmed: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    onClose: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

interface CopyProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
}

const ConfirmDialog: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();

    const copyProps: CopyProps = {};
    for (const key of Object.keys(props)) {
        if (ignoreKeys.indexOf(key) === -1) {
            copyProps[key] = props[key];
        }
    }

    const msgId = 'MUBP.ConfirmDialog';
    const msgIdTitle = msgId + 'Title';
    const msgIdCancel = msgId + 'Cancel';
    const msgIdOk = msgId + 'Ok';
    return (
        <Dialog
            {...copyProps}
            open={props.open}
            onClose={props.onClose}
            aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description"
            data-testid={msgId}
        >
            <DialogTitle id="alert-dialog-title" data-testid={msgIdTitle}>
                {formatMessage({ id: msgIdTitle, defaultMessage: 'Are you sure?' })}
            </DialogTitle>
            <DialogActions>
                <MyButton
                    label={formatMessage({ id: msgIdCancel, defaultMessage: 'Cancel' })}
                    onClick={props.onClose}
                    data-testid={msgIdCancel}
                />
                <MyButton
                    label={formatMessage({ id: msgIdOk, defaultMessage: 'OK' })}
                    onClick={props.onConfirmed}
                    color="primary"
                    autoFocus
                    data-testid={msgIdOk}
                />
            </DialogActions>
        </Dialog>
    );
};
export default ConfirmDialog;
