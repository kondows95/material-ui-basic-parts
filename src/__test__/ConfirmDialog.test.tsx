import React from 'react';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import ConfirmDialog from '../ConfirmDialog';

let tLib: TestLib;
let open = true;
let handleConfirmed: jest.Mock;
let handleClose: jest.Mock;

beforeEach(() => {
    open = true;
    handleConfirmed = jest.fn(() => {
        open = false;
    });
    handleClose = jest.fn(() => {
        open = false;
    });
});

describe('Basic Usage', () => {
    const langMessages = {
        'MUIBP.ok': 'MyOK',
        'MUIBP.cancel': 'MyCancel',
        'MUIBP.confirmDialogMessage': 'You can set message!',
    };
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={langMessages}>
                    <ConfirmDialog open={open} onConfirmed={handleConfirmed} onClose={handleClose} />
                </IntlProvider>
            )
        );
    });

    it('You can handle onClose', () => {
        tLib.click('confirm-cancel');
        expect(handleClose).toHaveBeenCalled();
        expect(open).toBeFalsy();
    });

    it('You can handle onConfirmed', () => {
        tLib.click('confirm-ok');
        expect(handleConfirmed).toHaveBeenCalled();
        expect(open).toBeFalsy();
    });

    it('You can set lables by Intl', () => {
        expect(tLib.get('confirm-ok').textContent).toBe('MyOK');
        expect(tLib.get('confirm-cancel').textContent).toBe('MyCancel');
        expect(tLib.get('alert-dialog-title').textContent).toBe('You can set message!');
    });
});

describe('Expansion Usage 1', () => {
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en">
                    <ConfirmDialog open={open} onConfirmed={handleConfirmed} onClose={handleClose} />
                </IntlProvider>
            )
        );
    });

    it('If Intl messages undefined', () => {
        expect(tLib.get('confirm-ok').textContent).toBe('OK');
        expect(tLib.get('confirm-cancel').textContent).toBe('Cancel');
        expect(tLib.get('alert-dialog-title').textContent).toBe('Are you sure?');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});
