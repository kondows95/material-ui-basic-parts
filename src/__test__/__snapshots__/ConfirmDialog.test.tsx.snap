// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expansion Usage 1 Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-504 makeStyles-dialog-502"
  />
</DocumentFragment>
`;
